src/main/scala/Game.scala

```diff
@@ -56,73 +54,85 @@ abstract class Game(
           uci.promotion.map(Role.FairySFPromotableRole),
           metrics
-        ) map { case (ncg, move) =>
-          ncg -> move
-        }
-      case Uci.SamuraiMove(uci)      =>
+        )
+      case Uci.SamuraiMove(uci)                         =>
         apply(
           Pos.Samurai(uci.orig),
           Pos.Samurai(uci.dest),
           promotion = None,
           metrics
-        ) map { case (ncg, move) =>
-          ncg -> move
-        }
-      case Uci.TogyzkumalakMove(uci) =>
+        )
+      case Uci.TogyzkumalakMove(uci)                    =>
         apply(
           Pos.Togyzkumalak(uci.orig),
           Pos.Togyzkumalak(uci.dest),
           promotion = None,
           metrics
-        ) map { case (ncg, move) =>
-          ncg -> move
-        }
-      case Uci.ChessDrop(uci)        =>
+        )
+      case Uci.BackgammonMove(uci)                      =>
+        apply(
+          Pos.Backgammon(uci.orig),
+          Pos.Backgammon(uci.dest),
+          promotion = None,
+          metrics
+        )
+      case Uci.ChessDrop(uci)                           =>
         drop(
           Role.ChessRole(uci.role),
           Pos.Chess(uci.pos),
           metrics
-        ) map { case (ncg, drop) =>
-          ncg -> drop
-        }
-      case Uci.FairySFDrop(uci)      =>
+        )
+      case Uci.FairySFDrop(uci)                         =>
         drop(
           Role.FairySFRole(uci.role),
           Pos.FairySF(uci.pos),
           metrics
-        ) map { case (ncg, drop) =>
-          ncg -> drop
-        }
-      case Uci.GoDrop(uci)           =>
+        )
+      case Uci.GoDrop(uci)                              =>
         drop(
           Role.GoRole(uci.role),
           Pos.Go(uci.pos),
           metrics
-        ) map { case (ncg, drop) =>
-          ncg -> drop
-        }
-      case Uci.GoPass(_)             =>
-        pass(
+        )
+      case Uci.BackgammonDrop(uci)                      =>
+        drop(
+          Role.BackgammonRole(uci.role),
+          Pos.Backgammon(uci.pos),
+          metrics
+        )
+      case Uci.BackgammonLift(uci)                      =>
+        lift(
+          Pos.Backgammon(uci.pos),
           metrics
-        ) map { case (ncg, pass) =>
-          ncg -> pass
-        }
-      case Uci.GoSelectSquares(uci)  =>
+        )
+      case Uci.GoSelectSquares(uci)                     =>
         selectSquares(
           uci.squares.map(Pos.Go(_)),
           metrics
-        ) map { case (ncg, selectSquares) =>
-          ncg -> selectSquares
-        }
+        )
+      case Uci.ChessDiceRoll(uci)                       =>
+        diceRoll(
+          uci.dice,
+          metrics
+        )
+      case Uci.BackgammonDiceRoll(uci)                  =>
+        diceRoll(
+          uci.dice,
+          metrics
+        )
+      case Uci.GoPass(_)                                => pass(metrics)
+      // TODO Backgammon should this be passing metrics?
```

Comment:
given this is automatic atm maybe not, but when we have multipoint matches, the user could think for some time before rolling dice and therefore need move metrics same as the other actions.